erizo/src/erizo/media/ExternalOutput.cpp

```diff
@@ -154,26 +148,59 @@ void ExternalOutput::writeAudioData(char* buf, int len) {
 
   long long currentTimestamp = head->getTimestamp();  // NOLINT
   if (currentTimestamp - firstAudioTimestamp_ < 0) {
-      // we wrapped.  add 2^32 to correct this. We only handle a single wrap around
-      // since that's 13 hours of recording, minimum.
-      currentTimestamp += 0xFFFFFFFF;
+    // we wrapped.  add 2^32 to correct this.
+    // We only handle a single wrap around since that's 13 hours of recording, minimum.
+    ELOG_DEBUG("Audio Wrapped: current: %lld, first: %lld", currentTimestamp, firstAudioTimestamp_);
+    currentTimestamp += 0xFFFFFFFF;
   }
 
-  long long timestampToWrite = (currentTimestamp - firstAudioTimestamp_) /  // NOLINT
-                                    (audio_stream_->codec->sample_rate / audio_stream_->time_base.den);
-  // generally 48000 / 1000 for the denominator portion, at least for opus
-  // Adjust for our start time offset
+  if (!SU.Slide(currentTimestamp, audio_measurements.rtcp_Packets, false)) {
+    ELOG_WARN("Slide Audio Failed, TS: %lld", currentTimestamp);
+    return;
+  }
 
-  // in practice, our timebase den is 1000, so this operation is a no-op.
-  timestampToWrite += audioOffsetMsec_ / (1000 / audio_stream_->time_base.den);
+  // Distance from rtp timestamp and the last rtcp timestamp
+  int delta_timestamp = currentTimestamp - audio_measurements.rtcp_Packets[0].getRtpTimestamp();
+  // Distance of the last rtcp packet and the first one
+  long long  addend = av_rescale_q(SU.NtpToMs(audio_measurements.rtcp_Packets[0].getNtpTimestampMSW(), //NOLINT
```

Comment:
here you can also use variable to store temporary values. It's hard to follow the code with them and you have to use lots of NOLINT because of line lengths.